### PR TITLE
Define and return PaymentMethod type, and getPaymentMethods method

### LINKED ISSUE #370: Property 'getPaymentMethods' does not exist on type 'AuthenticatedClient'
issue: If your using typescript, getPaymentMethods does not exist on the authenticated client, yet should be.

Steps to reproduce:
Simply create a typescript node project, connect to an authenticated client and call the method. You will get an error at compile time because the method is not defined on the Authenticated Client.

### PR DESCRIPTION
**Pull request by jfl209**
Fixes #370

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -124,6 +124,38 @@ declare module 'coinbase-pro' {
...
+    primary_buy: boolean,
+    primary_sell: boolean,
+    allow_buy: boolean,
+    allow_sell: boolean,
+    allow_deposit: boolean,
+    allow_withdraw: boolean
+    limits: PaymentMethodLimits
+  };
+
+  export type PaymentMethodLimit = {
```
Comment (blair) [inline index.d.ts:132]: Can you sort the definitions by use, so PaymentMethodLimitAmount first, then PaymentMethodLimit, PaymentMethodLimits and then PaymentMethod. That seems to be the style used in the rest of the file.

Comment (jfl209) [reply index.d.ts:132]: 👍

**Comment by jfl209** [comment]
It appears other methods are missing from the typescript declarations:

```
depositPayment
depositCrypto
withdrawPayment
createReport
getReportStatus
```

I'm going to add them to this pull request they are effectively the same (cause) as #370

**Comment by jfl209** [comment]
@blair How do I go about getting a merge?

**Comment by drewrothstein** [comment]
Hi, we are closing out PRs + Issues as this project is being archived.

**Pull request closed without merging**